boards/v2/intel/intel_ish/board.yml

```diff
@@ -0,0 +1,16 @@
+boards:
+
+  - name: intel_ish_5_4_1
+    vendor: intel
+    socs:
+      - name: intel_ish_5_4_1
+
+  - name: intel_ish_5_6_0
+    vendor: intel
+    socs:
+      - name: intel_ish_5_6_0
+
+  - name: intel_ish_5_8_0
+    vendor: intel
+    socs:
+      - name: intel_ish_5_8_0
```

Comment:
this should be using revisions

Reply:
but we don't have revisions for SoCs in Zephyr, only for boards, and those numbers seems to identify the SoC.

I assume somewhat similar to what nRF does with its package variant identification field, like `qfaa`, `ciaa`, etc.

So probably the cleanest approach would be to have a single board, and then identify the soc, something like:
```
board
 name: intel_ish
  vendor: intel
    socs:
      - name: intel_ish_5_4_1
      - name: intel_ish_5_6_0
      - name: intel_ish_5_8_0
```

but I would leave such decision at the discretion of the board maintainer.

Reply:
We do have board revisions which can select which soc is used, as per https://github.com/zephyrproject-rtos/zephyr/pull/67770/commits/6809aaefb987e5361355b676d9bf87dfe703ca60#diff-e26d2b274c8899adef8f82cb8fcf2a53a96838c4726caa9d7ad11c2afdb886af

Reply:
> but I would leave such decision at the discretion of the board maintainer.

@nashif , @jhedberg, @tbursztyka, @kwd-doodling - what do you think: should we keep ISH for HWMv2 as separate socs/boards for now same as currently, or migrate to board revisions, or  ?

Reply:
> We do have board revisions which can select which soc is used

I know, I was just pointing to the fact that `5_4_1`, `5_6_0`, `5_8_0` doesn't seem to be a revision of the board, but the revision of the SoC, and therefore using board revisions in this case might be a misnomer as well.

Reply:
> So probably the cleanest approach would be to have a single board, and then identify the soc, something like:
> ...
> but I would leave such decision at the discretion of the board maintainer.

I've changed ish_boards to variants - as a separate (last) commit in this PR.

```
 intel_ish_5_4_1 --> intel_ish/intel_ish_5_4_1
 intel_ish_5_6_0 --> intel_ish/intel_ish_5_6_0
 intel_ish_5_8_0 --> intel_ish/intel_ish_5_8_0
```

@nordicjm, @tejlmand, @nashif , @jhedberg, @tbursztyka, @kwd-doodling - is it what suits v.2 migration and the ISH design ?

Reply:
Not really no, it should be:
```
intel_ish@5.4.1
intel_ish@5.6.0
intel_ish@5.8.0
```

Reply:
In fact, they're different boards with different SoCs, not one board with different SoCs.
I've tried board revision. unfortunately it's a support of different boards with a same SoC, which doesn't fit our usage.

Reply:
> In fact, they're different boards with different SoCs, not one board with different SoCs. I've tried board revision. unfortunately it's a support of different boards with a same SoC, which doesn't fit our usage.

@kwd-doodling just to double check: does the new `board/soc/variant` naming scheme fit the ISH purposes ?,
i.e. what is proposed with the last commit here to have `intel_ish/intel_ish_5_4_1` etc. board names

Reply:
> In fact, they're different boards with different SoCs, not one board with different SoCs.

if they are in fact different boards, then I think they should still be modeled as individual boards.
I just had the impression that the `intel_ish` as a board was kind of a phony board, because the ish is a co-processor inside a SoC / SiP.

Reply:
@nashif what do you think about possible optimizations for ISH names ?

Reply:
> @nashif what do you think about possible optimizations for ISH names ?

@nashif, @kwd-doodling - are we going to rename SoC/boards somehow towards hwmv2 in this PR ?
If yes, then please suggest your new naming schema.